### PR TITLE
Add `is_potential_mixed_script_confusable_char` function

### PR DESCRIPTION
**Pull request by crlf0710**
<del>This is a prototype of the data required from mixed_script_confusable lint of rustc. I'm not really sure whether we need to give more detailed data to rustc for diagnostics. (e.g. This code point is potentially confusable with which code point or which script?)

Putting those aside, maybe we can have a early review first.</del>

Implements `is_potential_mixed_script_confusable_char` function.

A few other issues raised up during [adding the actual lint](https://github.com/rust-lang/rust/pull/72770) - #15 #16

### CODE REVIEW

**Comment by crlf0710** [comment]
There's a debug boolean value in the python script (search "debug = False", https://github.com/unicode-rs/unicode-security/pull/13/files#diff-c87c196441d88317a5ea3bf97e9fde0aR536), if anyone's curious why these codepoints are confusable, you can toggle it and regenerate the table file, which will include comments on why these code points are considered mixed script confusable.

**Comment by crlf0710** [comment]
The python calculation part is a little complex there. So i'll leave a brief description. The main idea is creating equivalence classes from `confusables.txt`, with the prototype as each equivalence class's representative element. Then compare each element pair within each equivalence class. If they're from different scripts, then mark each of them potentially mixed script confusable. And within that there're some special handling when the prototype has multiple code points.

File: scripts/unicode.py [hunk 1]
```diff
@@ -97,12 +141,244 @@ def load_confusables(f):
...
         for d_output in m.group(2).split():
-            d_outputitem = int(d_output, 16);
-            d_outputs.append(d_outputitem);
+            d_outputitem = int(d_output, 16)
+            d_outputs.append(d_outputitem)
         confusables.append((d_input, d_outputs))
 
     return confusables
 
+def aliases():
```
Comment (Manishearth) [inline scripts/unicode.py:162]: perhaps link to the original source in unicode-script so things can be manually updated if necessary

**Review by Manishearth — COMMENTED** [review]
It's really hard to review the python stuff, it would be nice if each function had a long comment explaining what its purpose is. currently i have to kinda figure that out from the code, and it's hard to review things when you don't know what they're doing

**Comment by crlf0710** [comment]
@Manishearth Will add some comments, thanks!

File: scripts/unicode.py (hunk 1 above)
Comment (crlf0710) [reply scripts/unicode.py:162]: Added comments below this line.

**Comment by crlf0710** [comment]
Added some comments. Also cc @pyfisch here.

**Comment by crlf0710** [comment]
Adjusted the APIs and added a very simple test.

File: scripts/unicode.py [hunk 2]
```diff
@@ -97,12 +141,280 @@ def load_confusables(f):
...
+            script_list.append(shortforms[script])
+        script_table.extend([(x, y, shortforms[script]) for (x, y) in scripts[script]])
+    script_list.sort()
+    script_table.sort(key=lambda w: w[0])
+    return (longforms, script_table)
+
+def is_script_ignored_in_mixedscript(source):
+    return source == 'Zinh' or source == 'Zyyy' or source == 'Zzzz'
+
+def process_mixedscript_single_to_multi(item_i, script_i, proto_lst, scripts):
```
Comment (Manishearth) [inline scripts/unicode.py:218]: what is this function trying to do? what does it operate on? where did those rules come from?

File: scripts/unicode.py [hunk 3]
```diff
@@ -97,12 +141,280 @@ def load_confusables(f):
...
+    # NotProcessed, DontAdd
+    return False, False
+
+def is_codepoint_identifier_allowed(c, identifier_allowed):
+    for data in identifier_allowed:
+        if c >= data[0] and c <= data[1]:
+            return True
+    return False
+
+def load_potential_mixedscript_confusables(f, identifier_allowed, scripts):
```
Comment (Manishearth) [inline scripts/unicode.py:280]: what this function returns, and what each argument is, should be documented.

**Review by Manishearth — APPROVED** [review]
Overall looks good, would appreciate docs on each function

File: scripts/unicode.py (hunk 3 above)
Comment (crlf0710) [reply scripts/unicode.py:280]: ok, i will add more comments.

File: scripts/unicode.py (hunk 2 above)
Comment (crlf0710) [reply scripts/unicode.py:218]: I will try to explain in comments.

**Comment by crlf0710** [comment]
Added more comments to document the details.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**